File: app/src/testFixtures/kotlin/com/example/hilt/AppModuleTest.java
package com.example.hilt;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import dagger.hilt.components.SingletonComponent;
import dagger.hilt.testing.TestInstallIn;

@Module
@TestInstallIn(
        components = {SingletonComponent.class},
        replaces = {AppModule.class}
)
public class AppModuleTest {

    @Provides
    @Singleton
    @SomeString
    public String provideSomeString() {
        return "Test";
    }
}

File: app/src/testFixtures/kotlin/com/example/hilt/AppModuleTest.kt
package com.example.hilt

import dagger.Module
import dagger.Provides
import dagger.hilt.components.SingletonComponent
import dagger.hilt.testing.TestInstallIn
import javax.inject.Singleton

@Module
@TestInstallIn(
    components = [SingletonComponent::class],
    replaces = [AppModule::class]
)
object AppModuleTest {

    @Provides
    @Singleton
    @SomeString
    internal fun provideSomeString(): String {
        return "Test"
    }
}
